yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MultiServerConnectionFactory.java

```diff
@@ -110,6 +116,24 @@ private List<RelationalConnection> alternateConnections(URI connectPath) {
         }).collect(Collectors.toList());
     }
 
+    /**
+     * Increment and return the next connection's initialConnection number.
+     * This allows us to better distribute the connections positions as connections are created per query in the tests
+     * and thus connections with a single query should increment their position between connection creation or else they
+     * will always execute with the same initial connection.
+     * @return the next initial connection number to use
+     */
+    private int getNextConnectionNumber() {
+        switch (connectionSelectionPolicy) {
+            case DEFAULT:
+                return DEFAULT_CONNECTION;
+            case ALTERNATE:
+                return currentConnection.addAndGet(1) % (1 + alternateFactories.size());
```

Comment:
This doesn't look right. Why are you doing `1+` here. If you have 2 connections, and `initialConnection` is `0` I think this would return:
1. 0 + 1 % 3 -> 1
2. 1 + 1 % 3 -> 2
3. 2 + 1 % 3 -> 0
4. 3 + 1 % 3 -> 1

But `2` is not valid.
And returning 1 seems wrong.

But, if my reading is right, this would fail tests, so maybe I'm wrong.

It might be clearer with
```
currentConnection.getAndUpdate(i -> (i + 1) % alternateFactories.size());
```

Reply:
the `1` is to account for the primary connection. Normally, we would have 1 `defaultFactory` and 1 `alternateFactory` in the list. 
It is incrementing prematurely, though, so the right statement is:
`currentConnection.getAndAdd(1) % (1 + alternateFactories.size())`
or
`currentConnection.getAndUpdate(i -> (i + 1) % (1 + alternateFactories.size()))`

Reply:
The difference is that the `%` is either accounted for in the counter or after the fact - I thought it is nice that it actually count the entire number of connections created, but don't have strong opinion on that

Reply:
Added field for the total number of connections, hopefully this will make it easier to follow.
Changed the logic to modulo the current connection to be within the range of the number of factories

Reply:
Oh, yeah, that is where I was confused, the fact that there is the "default" and the alternate.